permuta/patterns/perm.py

```diff
@@ -745,96 +745,157 @@ def is_sum_decomposable(self) -> bool:
 
     sum_decomposable = is_sum_decomposable
 
-    def descents(self) -> Iterator[int]:
-        """Yield the 0-based descents of self.
+    def descents(self, step_size: Optional[int] = None) -> Iterator[int]:
+        """Yield the 0-based descents of self. If step size is specified it only yields
+        descents of that size, otherwise it yields all descents.
 
         Examples:
             >>> tuple(Perm((0, 1, 3, 2, 4)).descents())
             (2,)
+            >>> tuple(Perm((0, 1, 3, 2, 4)).descents(step_size=2))
+            ()
             >>> tuple(Perm((3, 2, 1, 0)).descents())
             (0, 1, 2)
+            >>> tuple(Perm((3, 2, 1, 0)).descents(step_size=2))
+            ()
             >>> tuple(Perm((0, 1, 2)).descents())
             ()
+            >>> tuple(Perm((0, 1, 2)).descents(step_size=2))
+            ()
         """
+        if not (isinstance(step_size, int) or step_size is None):
```

Comment:
We usually don't perform those type of check. We rely on the linter to prevent call with the wrong type of argument.